c2corg_api/views/sitemap_xml.py

```diff
@@ -0,0 +1,222 @@
+import functools
+import logging
+
+from c2corg_api import DBSession, caching
+from c2corg_api.caching import cache_sitemap_xml
+from c2corg_api.models.cache_version import CacheVersion
+from c2corg_api.models.document import Document, DocumentLocale
+from c2corg_api.models.route import ROUTE_TYPE, RouteLocale
+from c2corg_api.models.user_profile import USERPROFILE_TYPE
+from c2corg_api.views import cors_policy, etag_cache
+from c2corg_api.views.validation import create_int_validator, \
+    validate_document_type
+from c2corg_common.utils.caching import get_or_create
+from cornice.resource import resource, view
+from pyramid.httpexceptions import HTTPNotFound
+from sqlalchemy.sql.functions import func
+from math import ceil
+from datetime import date, datetime, timezone
+from slugify import slugify
+
+log = logging.getLogger(__name__)
+
+# Search engines accept not more than 50000 urls per sitemap,
+# and the sitemap files may not exceed 10 MB. With 50000 urls the sitemaps
+# are not bigger than 9MB, but to be safe we are using 45000 urls per sitemap.
+# see http://www.sitemaps.org/protocol.html
+PAGES_PER_SITEMAP = 45000
+
+
+UI_ENTRY_POINTS = {
+    'a': 'areas',
+    'b': 'books',
+    'c': 'articles',
+    'i': 'images',
+    'm': 'maps',
+    'o': 'outings',
+    'r': 'routes',
+    'w': 'waypoints',
+    'x': 'xreports'
+}
+
+validate_page = create_int_validator('i')
+
+
+@resource(
+    collection_path='/sitemaps.xml', path='/sitemaps.xml/{doc_type}/{i}.xml',
```

Comment:
`sitemap.xml` instead of `sitemaps.xml`?

Reply:
I've stick to api logic : /routes and /routes/12345

Reply:
Depends what is referenced in `robots.txt`

Reply:
In robots.txt, you will have a reference to sitemap index, which is a list of site maps files. This one is on plural form.

The question can be asked for real sitemaps, between this four form : 

1. `sitemaps.xml/b/0.xml`
2. `sitemaps/b/0.xml`
3. `sitemap.xml/b/0.xml`
4. `sitemap/b/0.xml`

1) strictly stick to API route naming logic 
2) stick to API logic a little less, by removing useless extension in first 
3) make no sense :) 
4) the best IMO
 
All that said, I think it's a detail :)

Reply:
Yes, it is.